src/main/java/su/litvak/chromecast/api/v2/ChromeCast.java

```diff
@@ -86,6 +86,21 @@ public String getApplication() {
     public void setApplication(String application) {
         this.application = application;
     }
+    
+    /** Returns the {@link #channel} and opens it if it's not yet or
+     * no longer open.
+     * @return an open channel.
+     */
+    private Channel channel() throws IOException {
```

Comment:
The original intention of the change was to simplify normal usage. The current code throws NullPointerExceptions when you forget to `connect` in certain GETters. It's difficult for API users to know which getter does network I/O and which not. 

Most other comparable APIs I know (i.e. URLConnection) automatically connect on demand.

Reconnecting would be a completely other story.

I suggest we drop the PR, I understand your point for behaviour consistency (and catching NullPointerExceptions _are_ behaviour) and I'm not religious about this.

Thanks for your explanaition :+1: